com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -1226,21 +1214,48 @@ private boolean shouldBuildPlugins() {
                 String message = null;
                 Throwable exception = null;
                 boolean abort = false;
+                Map<IResource, String> outputResourceToCacheKey = new HashMap<IResource, String>();
                 try {
-                    builder.build(task);
-                    monitor.worked(1);
-                    for (IResource r : task.getOutputs()) {
-                        state.putSignature(r.getAbsPath(), taskSignature);
+                    // check if all output resources exist in the resource cache
+                    boolean allResourcesCached = true;
+                    for (IResource r : outputResources) {
+                        final String key = ResourceCacheKey.calculate(task, options, r);
+                        outputResourceToCacheKey.put(r, key);
+                        if (!r.isCacheable()) {
+                            allResourcesCached = false;
```

Comment:
No. For each output resource we also calculate the cache key and store that in the `outputResourceToCacheKey` map. This is done so that we don't have to do it in the next step when we iterate all output resources again to either get them from the cache or to build and put them in the cache.

I haven't actually measured the time it takes to calculate the cache key on a resource, but I can imagine that it is fairly costly as it calculates checksums on file content etc